### PR TITLE
fix(infiniteHits): do not cache the cached hits

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

This PR fixes the bug where InfiniteHits stored the cached hits, so clearing cache didn't make any effect.

* related
  * https://github.com/algolia/react-instantsearch/issues/2742
  * https://github.com/algolia/instantsearch.js/pull/4534

**Result**

InfiniteHits reads cache object everytime and do not store it inside the connector itself.

### CODE REVIEW

File: packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js [hunk 1]
```diff
@@ -80,14 +80,12 @@ export default createConnector({
 
     this._prevState = this._prevState || {};
 
-    const cache = props.cache || getInMemoryCache();
-    if (this._cachedHits === undefined) {
-      this._cachedHits = cache.read({ state: searchState }) || {};
-    }
+    this._cache = props.cache ? props.cache : this._cache || getInMemoryCache();
```
Comment (Haroenv) [inline packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js:83]: this is a very important fix!

**Review by Haroenv — APPROVED** [review]
I'm a fan, thanks!

File: packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js (hunk 1 above)
Comment (eunjae-lee) [reply packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js:83]: Yes, it especially makes sure to take `cache` prop into account even when user passes a different object!

Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js:83]: and makes sure we don't recreate the memory cache on every render

**Comment by alexpchin** [comment]
@Haroenv Do you know when this will be released?

**Comment by eunjae-lee** [comment]
@alexpchin It's just released in 6.10.0

**Pull request merged**